### PR TITLE
DeployUtil.deployFromJson to return Result<Deploy, Error>

### PR DESCRIPTION
**Pull request by zie1ony**
### Changed

- `DeployUtil.deployFromJson` returns now `Result<Deploy, Error>` instead of `Deploy | undefined`. `Error` has a `message` inside.

### CODE REVIEW

**Review by hoffmannjan — APPROVED** [review]

File: src/lib/DeployUtil.ts [hunk 1]
```diff
@@ -1144,13 +1144,32 @@ export const deployToJson = (deploy: Deploy) => {
...
+  }
+
+  const valid = validateDeploy(deploy);
+  if (valid.err) {
+    return new Err(new Error(valid.val));
+  }
+
+  if (!(deploy instanceof Deploy)) {
+    return new Err(new Error("'deploy' is not an instance of Deploy class."));
+  }
```
Comment (EdHastingsCasperAssociation) [inline src/lib/DeployUtil.ts]: Given this passed serialization into Deploy type above, is this necessary?

File: src/lib/DeployUtil.ts [hunk 2]
```diff
@@ -1144,13 +1144,32 @@ export const deployToJson = (deploy: Deploy) => {
...
+  try {
+    const serializer = new TypedJSON(Deploy);
+    deploy = serializer.parse(json.deploy);
+  } catch (serializationError) {
+    return new Err(serializationError);
   }
-  return undefined;
+
+  if (deploy === undefined || deploy === null) {
+    return Err(new Error("Can't parse to JSON"));
```
Comment (EdHastingsCasperAssociation) [inline src/lib/DeployUtil.ts]: is this the correct error message? it was _passed_ as JSON, it failed to parse as a `Deploy`.

File: src/lib/DeployUtil.ts [hunk 3]
```diff
@@ -1144,13 +1144,32 @@ export const deployToJson = (deploy: Deploy) => {
...
  * @param json
  */
-export const deployFromJson = (json: any) => {
-  const serializer = new TypedJSON(Deploy);
-  const deploy = serializer.parse(json.deploy);
-  if (deploy !== undefined && validateDeploy(deploy).ok) {
-    return deploy;
+export const deployFromJson = (json: any): Result<Deploy, Error> => {
+  if (json.deploy === undefined) {
+    return new Err(new Error("The Deploy JSON doesn't have 'deploy' key."));
```
Comment (EdHastingsCasperAssociation) [inline src/lib/DeployUtil.ts]: I haven't seen a sample of the json snippet, but I would assume that deploy would be a `field` not a `key` within that snippet?

**Review by EdHastingsCasperAssociation — APPROVED** [review]
see nitpicks

File: src/lib/DeployUtil.ts (hunk 3 above)
Comment (zie1ony) [reply src/lib/DeployUtil.ts]: Done

File: src/lib/DeployUtil.ts (hunk 2 above)
Comment (zie1ony) [reply src/lib/DeployUtil.ts]: Done

File: src/lib/DeployUtil.ts (hunk 1 above)
Comment (zie1ony) [reply src/lib/DeployUtil.ts]: I have added it to be extra, extra safe. When the code is transpiled to JS it strips all type checks. I talked to @hoffmannjan and he recommended to move it to `validateDeploy`, which I did.

**Pull request merged**